plugins/importexport/onix30/Onix30ExportPlugin.inc.php

```diff
@@ -101,18 +102,50 @@ function display($args, $request) {
 				]);
 				$templateMgr->display($this->getTemplateResource('index.tpl'));
 				break;
+			case 'exportSubmissionsBounce':
+				if (!$request->checkCSRF()) throw new Exception('CSRF mismatch!');
+				$json = new JSONMessage(true);
+				$json->setEvent('addTab', array(
+					'title' => __('plugins.importexport.native.results'),
+					'url' => $request->url(null, null, null, array('plugin', $this->getName(), 'export'), array('selectedSubmissions' => $request->getUserVar('selectedSubmissions'), 'csrfToken' => $request->getSession()->getCSRFToken())),
+				));
+				header('Content-Type: application/json');
+				return $json->getString();
 			case 'export':
+				$onixDeployment = new Onix30ExportDeployment($context, $user);
+
 				$exportXml = $this->exportSubmissions(
 					(array) $request->getUserVar('selectedSubmissions'),
-					$request->getContext(),
-					$request->getUser()
+					$context,
+					$user,
+					$onixDeployment
 				);
+
+				$problems = $onixDeployment->getWarningsAndErrors();
+				$foundErrors = $onixDeployment->isProcessFailed();
+
+				if ($exportXml) {
+					$path = $this->writeExportedFile($exportXml, $onixDeployment->getContext());
+					$templateMgr->assign('exportPath', $path);
+				}
+
+				$templateMgr->assign('validationErrors', $onixDeployment->getXMLValidationErrors());
+
+				$templateMgr->assign('errorsAndWarnings', $problems);
+				$templateMgr->assign('errorsFound', $foundErrors);
+				$templateMgr->assign('onixPlugin', $this);
+
+				$json = new JSONMessage(true, $templateMgr->fetch($this->getTemplateResource('resultsExport.tpl')));
+				header('Content-Type: application/json');
+				return $json->getString();
+			case 'downloadExportFile':
+				$downloadPath = $request->getUserVar('downloadFilePath');
```

Comment:
This is an arbitrary "download a file then delete it" feature, I think :scream:

Reply:
You are right @asmecher.. The user can "play" with the hidden field to download what ever (and delete it afterwards).

I am proposing a fix for that, though not sure if its the correct approach: The code is now checking if the file that needs to be downloaded is in the "correct" export path, and also if the name matches the name that the plugin expects to be.

Reply:
@defstat, if the handling code already knows the correct export filename, then why do we need to bounce it off the user in the first place? Are there several possible valid filenames?

Reply:
Thanks @asmecher ... The code now passes only the date part of the file in order to determine the file that will be downloaded